### PR TITLE
fix(refactor): fix comments after review

### PR DESCRIPTION
**Pull request by HoangLiarPhuongLe**
1. Fixed
- Image error: fixed ( by deleting )
- There is a gap between error message and input
- Fixed the error that input amount was slanted upwards when show error message
- Rename helpers.js file to format.js
- Move localStorage to helpers folder
- Split template.js file into 2 files templateTransaction and templateSummary
- Update file Readme.md
- Remove commas between transactions
2. Haven't fixed yet
- Show full error message for inputs when successfully adding transaction for the first time

### CODE REVIEW

File: src/javascripts/templates/templateTransaction.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
-import { formatDate, formatNumber } from "../helpers/helpers"
+import { formatDate, formatNumber } from "../helpers/format"
 
-class Template {
+class TemplateTransaction {
```
Comment (phuocduong-agilityio) [inline src/javascripts/templates/templateTransaction.js]: Should be TransactionTemplate

File: src/javascripts/views/homeView.js [hunk 2]
```diff
@@ -51,7 +52,7 @@ class HomeView {
     }
 
     renderCategory(category, transactions, totalOutFlow) {
-       const categoryTemplate = Template.displayCategory(category, transactions, totalOutFlow);
+       const categoryTemplate = TemplateTransaction.displayCategory(category, transactions, totalOutFlow);
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/homeView.js]: After const, add empty line

File: src/styles/components/popup.css [hunk 3]
```diff
@@ -147,3 +148,16 @@
   font-weight: 500;
   border: none;
 }
+.error-messages {
```
Comment (phuocduong-agilityio) [inline src/styles/components/popup.css]: 1. Format
2. Error is not related popup

File: src/javascripts/helpers/validationForm.js [hunk 4]
```diff
@@ -61,8 +61,8 @@ export const validateTransaction = (data) => {
   ]
 
   for(const field of transactionFields){
-   
-    if (!data[field.field] && !data[field.field].trim()){
+
```
Comment (phuocduong-agilityio) [inline src/javascripts/helpers/validationForm.js]: Remove this

File: src/javascripts/templates/templateSummary.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+     /**
+     * Constructor of TemplateSummary object
+     */
+     constructor(){ }
+
+     /**
+     * HTML TemplateSummary for render an Summary object
+     * @param {object} summary 
+     */
+
```
Comment (phuocduong-agilityio) [inline src/javascripts/templates/templateSummary.js]: Remove this

Comment (HoangLiarPhuongLe) [reply src/javascripts/templates/templateSummary.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks !

File: src/javascripts/helpers/validationForm.js (hunk 4 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/helpers/validationForm.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks !

File: src/styles/components/popup.css (hunk 3 above)
Comment (HoangLiarPhuongLe) [reply src/styles/components/popup.css]: I fixed @phuocduong-agilityio . Please help me check. Thanks !

File: src/javascripts/views/homeView.js (hunk 2 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/views/homeView.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks !

File: src/javascripts/templates/templateTransaction.js (hunk 1 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/templates/templateTransaction.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks !

File: src/javascripts/templates/summaryTemplate.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+     /**
+     * Constructor of TemplateSummary object
+     */
+     constructor(){ }
+
+     /**
+     * HTML TemplateSummary for render an Summary object
+     * @param {object} summary 
+     */
+
```
Comment (phuocduong-agilityio) [inline src/javascripts/templates/summaryTemplate.js]: Remove empty line here

Comment (HoangLiarPhuongLe) [reply src/javascripts/templates/summaryTemplate.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks !

File: src/javascripts/templates/transactionTemplate.js [hunk 7]
```diff
@@ -19,36 +18,36 @@ class Template {
...
           </div>
-        </div>
+        </li>
         `;
     }
 
     static displayCategory = (category, transactions, totalOutFlow) => { 
       const isIncome = totalOutFlow >= 0;
       const totalOutFlowSign = isIncome ? '+' : '-';
+      const transaction = transactions.map(transaction => TransactionTemplate.displayTransaction(transaction));
```
Comment (phuocduong-agilityio) [inline src/javascripts/templates/transactionTemplate.js]: Please choose another name, map return array, not item

File: src/javascripts/views/homeView.js [hunk 8]
```diff
@@ -51,7 +52,8 @@ class HomeView {
     }
 
     renderCategory(category, transactions, totalOutFlow) {
-       const categoryTemplate = Template.displayCategory(category, transactions, totalOutFlow);
+       const categoryTemplate = TransactionTemplate.displayCategory(category, transactions, totalOutFlow);
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/homeView.js]: IMO, display mean you will show it on UI. But in this case, you create template, not display

Comment (HoangLiarPhuongLe) [reply src/javascripts/views/homeView.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks

File: src/javascripts/templates/transactionTemplate.js (hunk 7 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/templates/transactionTemplate.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks

**Pull request merged**